### ISSUE TITLE
%matplotlib magic: print a list of available backends

### ISSUE THREAD

**Issue by cel4** [ipython/ipython#8444]
Is there an easy way to print a list of all available backends? Starting the interpreter with `ipython --matplotlib foo` prints an error message with all available backends. Is there something similar for the `%matplotlib` magic? Wouldn't it be helpful to have something like `%matplotlib --list` to print a list of all available backends?

**Comment by Carreau** [comment]
I don't see any harm in having that. `%matplotlib?` give the list at the bottom already.

**Comment by Carreau** [comment]
Do you want to try a PR ?

**Comment by cel4** [comment]
I just wanted to leave the idea here - I have very little time at the moment. But this is not urgent - so I may put together a PR at some time.

**Comment by koenvb** [comment]
Mind if I give it a try. I am quite a beginner but I assume this is not so hard to do?

**Comment by takluyver** [comment]
It shouldn't be too difficult, and we're quite happy whoever implements it. :-)

The code for the `%matplotlib` and `%pylab` magics is in `IPython.core.magics.pylab`. If you want to know anything else, feel free to ask us, either here on our gitter chat room at https://gitter.im/ipython/ipython

**Comment by koenvb** [comment]
I added something for the %matplotlib magic

```
In [3]: %matplotlib -l
Available matplotlib backends: {'osx': 'MacOSX', 'qt4': 'Qt4Agg', 'qt5': 'Qt5Agg', 'gtk3': 'GTK3Agg', 'notebook': 'nbAgg', 'wx': 'WXAgg', 'qt': 'Qt4Agg', 'nbagg': 'nbAgg', 'gtk': 'GTKAgg', 'tk': 'TkAgg', 'inline': 'module://ipykernel.pylab.backend_inline'}

In [4]: %matplotlib --list
Available matplotlib backends: {'osx': 'MacOSX', 'qt4': 'Qt4Agg', 'qt5': 'Qt5Agg', 'gtk3': 'GTK3Agg', 'notebook': 'nbAgg', 'wx': 'WXAgg', 'qt': 'Qt4Agg', 'nbagg': 'nbAgg', 'gtk': 'GTKAgg', 'tk': 'TkAgg', 'inline': 'module://ipykernel.pylab.backend_inline'}

```

**Comment by Carreau** [comment]
Closed by #8449 and #8506
